base/src/main/java/com/smartdevicelink/proxy/SystemCapabilityManager.java

```diff
@@ -450,16 +450,17 @@ private boolean isSubscribedToSystemCapability(SystemCapabilityType systemCapabi
 	 */
 	private Object getCapabilityPrivate(final SystemCapabilityType systemCapabilityType, final OnSystemCapabilityListener scListener, final Boolean subscribe, final boolean forceUpdate) {
 		Object cachedCapability = cachedSystemCapabilities.get(systemCapabilityType);
-		OnSystemCapabilityListener listener = scListener;
 
-		if (!forceUpdate && cachedCapability != null && listener != null) {
-			listener.onCapabilityRetrieved(cachedCapability);
-			listener = null;
+		boolean shouldUpdateSystemCapabilitySubscription = (subscribe != null) && (subscribe != isSubscribedToSystemCapability(systemCapabilityType)) && supportsSubscriptions();
+		boolean shouldSendGetCapabilityRequest = forceUpdate || (cachedCapability == null) || shouldUpdateSystemCapabilitySubscription;
+		boolean shouldCallListenerWithCachedValue = (cachedCapability != null) && (scListener != null) && !shouldSendGetCapabilityRequest;
```

Comment:
1. Currently, iOS calls the listener with a cached value of `nil`. This may be a point of alignment needed.
2. Why does it not call with the cached value if we are sending a `GetCapability` request? Should the `&&` there be an `||`?